File: Editorial/1901-2000/1935_Maximum_Number_of_Words_You_Can_Type/TypeScript/1_Approach_Traversal_Hash_Table.ts
function canBeTypedWords(text: string, brokenLetters: string): number {
    const broken = new Set<string>(); // set of broken letter keys
    for (const ch of brokenLetters) {
        broken.add(ch);
    }
    let res = 0; // the number of words that can be fully inputted
    let flag = true; // is the current character in the word completely inputtable
    for (const ch of text) {
        if (ch === " ") {
            // the current character is a space, check the status of the previous word, update the count and initialize the flag
            if (flag) {
                ++res;
            }
            flag = true;
        } else if (broken.has(ch)) {
            // the current character cannot be entered, the word it is in cannot be fully entered, update flag
            flag = false;
        }
    }
    // judge the status of the last word and update the count
    if (flag) {
        ++res;
    }
    return res;
}

File: Editorial/1901-2000/1935_Maximum_Number_of_Words_You_Can_Type/C/1_Approach_Traversal_Hash_Table.c
int canBeTypedWords(char* text, char* brokenLetters) {
    bool broken[26] = {false};  // set of broken letter keys
    for (int i = 0; brokenLetters[i]; i++) {
        broken[brokenLetters[i] - 'a'] = true;
    }
    int res = 0;  // the number of words that can be fully inputted
    bool flag =
        true;  // is the current character in the word completely inputtable
    for (int i = 0; text[i]; i++) {
        if (text[i] == ' ') {
            // the current character is a space, check the status of the
            // previous word, update the count and initialize the flag
            if (flag) {
                ++res;
            }
            flag = true;
        } else if (broken[text[i] - 'a']) {
            // the current character cannot be entered, the word it is in cannot
            // be fully entered, update flag
            flag = false;
        }
    }
    // judge the status of the last word and update the count
    if (flag) {
        ++res;
    }
    return res;
}
